cli/src/commands/workspace.rs

```diff
@@ -279,6 +282,76 @@ fn cmd_workspace_root(
     Ok(())
 }
 
+fn create_and_check_out_recovery_commit(
+    command: &CommandHelper,
+    ui: &mut Ui,
+    workspace: Workspace,
+) -> Result<Arc<ReadonlyRepo>, CommandError> {
+    let operation = command.resolve_operation(ui, workspace.repo_loader())?;
+    let repo = workspace.repo_loader().load_at(&operation)?;
+    let workspace_id = workspace.workspace_id().clone();
+    let tree_id = repo.store().empty_merged_tree_id();
+
+    let mut workspace_command = command.for_loaded_repo(ui, workspace, repo.clone())?;
+    let (mut locked_workspace, commit) =
+        workspace_command.unchecked_start_working_copy_mutation()?;
+    let commit_id = commit.id();
+
+    let mut tx = repo.start_transaction(command.settings());
+    let mut_repo = tx.mut_repo();
+    let new_commit = mut_repo
+        .new_commit(command.settings(), vec![commit_id.clone()], tree_id.clone())
+        .write()?;
+    mut_repo.set_wc_commit(workspace_id, new_commit.id().clone())?;
+    let ret = Ok(tx.commit("recovery commit"));
+
+    locked_workspace.locked_wc().reset_to_empty()?;
+    locked_workspace.finish(operation.id().clone())?;
```

Comment:
Since `tx.commit()` creates new operation, its id should be recorded instead.
```rust
    let repo = tx.commit("recovery commit");
    ...
    locked_workspace.finish(repo.op_id().clone())?;
```

The operation id recorded here will be overwritten by `maybe_snapshot()` though.

Reply:
Ah, thanks for noticing. I've changed it.